src/Collection/CollectionTrait.php

```diff
@@ -86,7 +86,11 @@ public function reject(callable $c)
      */
     public function every(callable $c)
     {
-        foreach ($this->unwrap() as $key => $value) {
+        $iterator = $this->unwrap();
+        if (count($iterator) === 0) {
+            return false;
+        }
+        foreach ($iterator as $key => $value) {
```

Comment:
better set a boolean here as the first line of this loop. If the boolean is still false at the end of the loop, that means the iterator was empty